### PR TITLE
remove unused code

### PR DESCRIPTION
**Pull request by keakon**

### CODE REVIEW

File: starlette/staticfiles.py [hunk 1]
```diff
@@ -145,7 +145,6 @@ def get_path(self, scope: Scope) -> str:
     async def lookup_path(
         self, path: str
     ) -> typing.Tuple[str, typing.Optional[os.stat_result]]:
-        stat_result = None
```
Comment (lovelydinosaur) [inline starlette/staticfiles.py:148]: Yup.

File: starlette/staticfiles.py [hunk 2]
```diff
@@ -27,7 +27,7 @@ class NotModifiedResponse(Response):
     )
 
     def __init__(self, headers: Headers):
-        return super().__init__(
```
Comment (lovelydinosaur) [inline starlette/staticfiles.py:30]: Yup.

File: starlette/routing.py [hunk 3]
```diff
@@ -161,9 +161,9 @@ def __init__(
         if methods is None:
             self.methods = None
         else:
-            self.methods = set([method.upper() for method in methods])
+            self.methods = set(method.upper() for method in methods)
             if "GET" in self.methods:
-                self.methods |= set(["HEAD"])
```
Comment (lovelydinosaur) [inline starlette/routing.py:167]: Yeah, okay - prefer your version of these two lines.

File: starlette/middleware/wsgi.py [hunk 4]
```diff
@@ -93,7 +94,7 @@ def __init__(self, app: typing.Callable, scope: Scope) -> None:
                     self.exc_info[1], self.exc_info[2]
                 )
         finally:
-            if not sender.done():
```
Comment (lovelydinosaur) [inline starlette/middleware/wsgi.py:96]: I don't see any value in the two lines changed here.

File: starlette/routing.py [hunk 5]
```diff
@@ -231,8 +231,6 @@ def __init__(
             # Endpoint is a class. Treat it as ASGI.
             self.app = endpoint
 
-        regex = "^" + path + "$"
-        regex = re.sub("{([a-zA-Z_][a-zA-Z0-9_]*)}", r"(?P<\1>[^/]+)", regex)
```
Comment (lovelydinosaur) [inline starlette/routing.py:236]: Any idea where these lines have been left over from? They don't make sense right now, but might be a good idea to understand how they came to be here?

File: starlette/middleware/wsgi.py (hunk 4 above)
Comment (keakon) [reply starlette/middleware/wsgi.py:96]: `self.loop.create_task()` may raise an exception, then `sender` will be referenced before assignment.

File: starlette/routing.py (hunk 5 above)
Comment (keakon) [reply starlette/routing.py:236]: This [commit](https://github.com/encode/starlette/commit/de71b8594807dae907eba636f01dc13fdb355f9d#diff-34fba745b50527bfb4245d02afd59246L122) removed the pattern in `Route`, but forgot `WebSocketRoute`.

File: starlette/middleware/wsgi.py (hunk 4 above)
Comment (lovelydinosaur) [reply starlette/middleware/wsgi.py:96]: Okay, then let's move `sender = ` to just outside the `try ... finally` block

**Comment by blueyed** [comment]
@keakon 
Ping.
Rebasing should fix the CI failure also.

**Comment by lovelydinosaur** [comment]
Looks good - thanks!

**Pull request merged**